src/nodes/access.ts

```diff
@@ -0,0 +1,212 @@
+// Code largely based on https://github.com/ajafff/tsutils
+// Original license: https://github.com/ajafff/tsutils/blob/26b195358ec36d59f00333115aa3ffd9611ca78b/LICENSE
+
+import ts from "typescript";
+
+import { isAssignmentKind } from "../syntax";
+
+/* eslint-disable perfectionist/sort-enums */
+/**
+ * What operations(s), if any, an expression applies.
+ */
+export enum AccessKind {
```

Comment:
Could it be better to express these in bitwise notation, to make it clear that's how they are supposed to be used?

```ts
export enum AccessKind {
  None = 0,
	Read = 1 << 0,
	Write = 1 << 1,
	Delete = 1 << 2,
	ReadWrite = Read | Write,
}
```

Reply:
It's probably a bit unnecessary though.

Reply:
I like it!